se_shared/include/se/path_planning/candidate_view.hpp

```diff
@@ -228,16 +228,13 @@ Eigen::Vector3i CandidateView<T>::getOffsetCandidate(const Eigen::Vector3i &cand
             << " offset by " << offset_v << " results in " << offset_cand_v.x() << " "
             << offset_cand_v.y() << " " << offset_cand_v.z() << " voxel state "
             << volume_._map_index->get(offset_cand_v).st;*/
+
   int free_sphere = collision_check.isSphereCollisionFree(offset_cand_v);
-  if (is_valid) {
-    if (free_sphere==1) {
+
+  if (is_valid && free_sphere == 1) {
```

Comment:
This structure would be faster when `is_valid == false` since `isSphereCollisionFree` will not be called:
```
if (is_valid) {
  const int free_sphere = collision_check.isSphereCollisionFree(offset_cand_v);
  if (free_sphere == 1) {
```